### ISSUE TITLE
Types for graphics

### ISSUE THREAD

**Issue by mwiebe** [libdynd/libdynd#1246]
An issue to compile information about graphics-related types.

* http://graphics.pixar.com/usd/docs/api/_usd__page__datatypes.html#Usd_Basic_Datatypes
* http://www.lighthouse3d.com/tutorials/glsl-12-tutorial/data-types-and-variables/
* https://renderman.pixar.com/resources/current/RenderMan/types.html

There's a naming problem for the parameterizations of vector size and data size. USD mixes them a bit, e.g. has `int64` as a 64-bit signed integer, and `int3` as a 3-vector of 32-bit integers. It also uses `double3`, `float3`, and `half3` as the various precisions of floating point 3-vectors. Using the types `3 * float64`, `3 * float32`, and `3 * float16` doesn't quite feel right, because they should generally be treated as scalar units, e.g. a 1-dimensional array of 3-vectors and a grey scale 2 dimensional image of height 3 shouldn't have the same type.

**Comment by izaid** [comment]
Yeah, CUDA also does this -- double3, double2, float3, float2, etc...
